### PR TITLE
Cart page management

### PR DESCRIPTION
**Pull request by Manju-sree**
### Cart page
I have added the Cart page. It's still in the experimental phase.(as place order is not functioning).
- Added cart page functionality, where you find Total Amount calculation after discount  in TotalPriceCard. 
- Add item to wishlist from cart and removing that item from cart is managed in HorizontalCard of cart page.

**What not to check**:
-  Responsiveness
-  menu functionality
- Authentication

Preview: https://deploy-preview-9--angry-dubinsky-3738ba.netlify.app/Cart

### CODE REVIEW

File: src/components/TotalPriceCard/TotalPriceCard.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            <div className="bill-details-row">
+                <div className="col-left-wd ">Price ({cartCount} items)</div>
+                <div className="col-right-wd  text-align-right">
+                    ₹{totalPriceAmount}
+                </div>
+            </div>
+            <div className="bill-details-row">
+                <div className="col-left-wd ">Discount </div>
+                <div className="col-right-wd  text-align-right primary-color">
+                    {discountedPrice}
```
Comment (JantuDeb) [inline src/components/TotalPriceCard/TotalPriceCard.jsx]: add ₹ rupee symbol here also

File: src/components/TotalPriceCard/TotalPriceCard.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    // calculating total price items
+    const totalPriceAmount = cartItem.reduce(
+        (acc, currentItem) => currentItem.qty * currentItem.price + acc,
+        0
+    );
+
+    const discountedPrice = cartItem.reduce(
+        (acc, currentItem) =>( currentItem.price * currentItem.discount)/100 *currentItem.qty+acc,
+        0
+    );
```
Comment (JantuDeb) [inline src/components/TotalPriceCard/TotalPriceCard.jsx:16]: You can do this with one `reduce` call
  ```
  const totalPriceAmount = cartItem.reduce(
        (acc, currentItem) => ({...acc, total:currentItem.qty * currentItem.price + acc.total, discount:( currentItem.price * currentItem.discount)/100 *currentItem.qty+acc.discount}),
       {total:0, discount:0}
    );
```

**Review by JantuDeb — APPROVED** [review]
Suggested a changes

File: src/pages/Cart/CartPage.jsx [hunk 3]
```diff
@@ -16,42 +17,22 @@ export const Cart = () => {
...
+              {cartItem && cartItem.map((item) => (
+                <HorizontalCard item={item} />
+              ))}
+            </li>
+          </ul>
+          <div className="cart-right-container">
+          {cartItem.length === 0 ? (
+              ""
+            ):(
+            <TotalPriceCard cartState={cartState}/>)}
```
Comment (rkapoor10) [inline src/pages/Cart/CartPage.jsx:32]: you can use && if you don't need the if part  like this
```(cartItem?.length === 0) && <TotalPriceCard cartState={cartState}/>```

**Review by rkapoor10 — APPROVED** [review]
The code is well structured and good to merge

File: src/components/TotalPriceCard/TotalPriceCard.jsx (hunk 1 above)
Comment (Manju-sree) [reply src/components/TotalPriceCard/TotalPriceCard.jsx]: ok done

File: src/components/TotalPriceCard/TotalPriceCard.jsx (hunk 2 above)
Comment (Manju-sree) [reply src/components/TotalPriceCard/TotalPriceCard.jsx:16]: wow, thanks for writing down this approach, but i need two different reducers , bcoz i'm displaying that two variables in UI.

File: src/pages/Cart/CartPage.jsx (hunk 3 above)
Comment (Manju-sree) [reply src/pages/Cart/CartPage.jsx:32]: yeah, thanks for suggestion, its basically showing pricecard if there is atleast one item in cart or else displaying nothing in pricecard details.

**Pull request merged**